internal/install/discovery/regex_process_filterer.go

```diff
@@ -23,45 +23,68 @@ func NewRegexProcessFilterer(r recipes.RecipeFetcher) *RegexProcessFilterer {
 	return &f
 }
 
-func (f *RegexProcessFilterer) filter(ctx context.Context, processes []types.GenericProcess) ([]types.GenericProcess, error) {
+func (f *RegexProcessFilterer) filter(ctx context.Context, processes []types.GenericProcess) ([]types.ProcessInfoWrap, error) {
+	processesInfo := getProcessesInfo(processes)
```

Comment:
Consider renaming `processesInfo` to `runningProcesses`, and returning a slice of `GenericProcess` from the method here?  We can resolve the `CmdLine()` value down below when we perform the matching.